### PR TITLE
Add session hijacking functionality to freebsd openbsd dragonfly netbsd and darwin platforms

### PR DESCRIPTION
**Pull request by ugjka**
* Use `github.com/ftrvxmtrx/fd` for sending an receiving file descriptors. It is a much more robust library that doesn't do dirty hacks
    
 * Add session hijacking functionality to freebsd openbsd dragonfly netbsd and darwin platforms. Linux hijacker has been cleaned up.
    
 * Recon_dummy.go is an empty hijacker that uses inverted build constraints for platforms other than linux and those defined in recon_unix.go

### CODE REVIEW

File: recon_linux.go [hunk 1]
```diff
@@ -4,67 +4,69 @@ import (
...
 
 // Attempt to hijack session previously running bot
-func (irc *Bot) hijackSession() bool {
-	unaddr, err := net.ResolveUnixAddr("unix", irc.unixastr) // The only way to get an error here is if the first parameter is not one of "unix", "unixgram" or "unixpacket". That will never happen.
-	if err != nil {
-		panic(err)
-	}
-	con, err := net.DialUnix("unix", nil, unaddr)
+func (bot *Bot) hijackSession() bool {
+	con, err := net.Dial("unix", bot.unixastr)
```
Comment (icholy) [inline recon_linux.go:52]: If you use [net.DialUnix](https://golang.org/pkg/net/#DialUnix)  you can remove the type assertion further down.

File: recon_linux.go [hunk 2]
```diff
@@ -4,67 +4,69 @@ import (
...
 	}
-	ncon, err := sendfd.RecvFD(con)
+	defer con.Close()
+
+	ncon, err := fd.Get(con.(*net.UnixConn), 1, nil)
 	if err != nil {
 		panic(err)
 	}
-	netcon, err := net.FileConn(ncon)
+	defer ncon[0].Close()
```
Comment (icholy) [inline recon_linux.go:63]: Is it possible for `len(ncon)` to be zero?

File: recon_linux.go (hunk 1 above)
Comment (ugjka) [reply recon_linux.go:52]: net.DialUnix can't use a string path it needs net.*UnixAddr which you can only get from net.ResolveUnixAddr, I think.

I find a simple dial and type assertion cleaner

Comment (icholy) [reply recon_linux.go:52]: Ah, that makes sense.

File: recon_linux.go (hunk 2 above)
Comment (ugjka) [reply recon_linux.go:63]: I don't think so, we'd get an error if there was nothing

**Pull request merged**